flexget/plugins/list/ombi_list.py

```diff
@@ -0,0 +1,540 @@
+            self.config.get('port'),
+            parsedurl.path,
+        )
+        data = {'username': self.config.get('username'), 'password': self.config.get('password')}
+        headers = {'Content-Type': 'application/json', 'Accept': 'application/json'}
+        try:
+            access_token = (
+                requests.post(url, json=data, headers=headers).json().get('access_token')
+            )
+            return access_token
+        except (RequestException, ValueError) as e:
+            raise plugin.PluginError('Ombi username and password login failed: %s' % e)
+
+    def ombi_auth(self) -> Dict[str, str]:
+        if self.config.get('api_key'):
+            log.debug('Authenticating via api_key')
+            api_key = self.config.get('api_key')
+            header = {'ApiKey': api_key}
+            return header
+        elif self.config.get('username') and self.config.get('password'):
+            log.debug('Authenticating via username: %s', self.config.get('username'))
+            access_token = self.get_access_token()
+            return {"Authorization": "Bearer %s" % access_token}
+        else:
+            raise plugin.PluginError(
+                'Error: an api_key or username and password must be configured'
+            )
+
+    def _get_base_url(self):
+        url = urlparse(self.config.get('base_url'))
+        port = self.config.get('port')
+
+        return f"{url.scheme}://{url.netloc}:{port}{url.path}"
+
+    def get_request_list(self):
+        auth_header = self.ombi_auth()
+
+        parsedurl = urlparse(self.config.get('base_url'))
+        url = ''
+        if self.config.get('type') in ['movies']:
+            url = '%s://%s:%s%s/api/v1/Request/movie' % (
+                parsedurl.scheme,
+                parsedurl.netloc,
+                self.config.get('port'),
+                parsedurl.path,
+            )
+        elif self.config.get('type') in ['shows', 'seasons', 'episodes']:
+            url = '%s://%s:%s%s/api/v1/Request/tv' % (
+                parsedurl.scheme,
+                parsedurl.netloc,
+                self.config.get('port'),
+                parsedurl.path,
+            )
+        else:
+            raise plugin.PluginError('Error: Unknown list type %s.' % (self.config.get('type')))
+        log.debug('Connecting to Ombi to retrieve request type : %s', self.config.get('type'))
+        try:
+            return requests.get(url, headers=auth_header).json()
+        except (RequestException, ValueError) as e:
+            raise plugin.PluginError('Unable to connect to Ombi at %s. Error: %s' % (url, e))
+
+    def generate_movie_entry(self, parent_request):
+
+        log.debug('Found: %s', parent_request.get('title'))
+
+        # If we dont have an imdb id, maybe we should get it from the title?
+        if not parent_request.get('imdbId'):
+            simdburl = ''
+        else:
+            simdburl = 'http://www.imdb.com/title/' + parent_request.get('imdbId') + '/'
```

Comment:
using variable for this would be a bit cleaner and make f-strings viable. Same thing a bit lower.